calico_containers/tests/st/test_base.py

```diff
@@ -20,3 +24,26 @@ def tearDown(self):
         containers = docker.ps("-qa").split()
         for container in containers:
             DockerHost.delete_container(container)
+
+    def start_etcd(self):
+        """
+        Starts a separate etcd container.
+        """
+
+        docker.run("-d",
+                   "-p", "2379:2379",
+                   "-p", "2380:2380",
+                   "--name", "etcd", "quay.io/coreos/etcd:v2.0.10",
+                   name="calico",
+                   advertise_client_urls="http://%s:2379" % self.ip,
+                   listen_client_urls="http://0.0.0.0:2379",
+                   initial_advertise_peer_urls="http://%s:2380" % self.ip,
+                   listen_peer_urls="http://0.0.0.0:2380",
+                   initial_cluster_token="etcd-cluster-2",
+                   initial_cluster="calico=http://%s:2380" % self.ip,
+                   initial_cluster_state="new",
+                  )
+
+    def get_ip(self):
```

Comment:
Note that '8.8.8.8' isn't a fake IP, it's one of Google's DNS servers, so
it's probably one of the most durable IPs in existance for pings. That
durability is why the examples for this approach tend to use 8.8.8.8. (If
you're really paranoid you can fall back to 8.8.4.4, or even some other
cloud provider's HA endpoint).

On Fri, May 29, 2015 at 10:39 AM, robbrockbank <notifications@github.com>
wrote:

> In calico_containers/tests/st/test_base.py
> <https://github.com/Metaswitch/calico-docker/pull/154#discussion_r31346709>
> :
>
> > +
> > +        docker.run("-d",
> > +                   "-p", "2379:2379",
> > +                   "-p", "2380:2380",
> > +                   "--name", "etcd", "quay.io/coreos/etcd:v2.0.10",
> > +                   name="calico",
> > +                   advertise_client_urls="http://%s:2379" % self.ip,
> > +                   listen_client_urls="http://0.0.0.0:2379",
> > +                   initial_advertise_peer_urls="http://%s:2380" % self.ip,
> > +                   listen_peer_urls="http://0.0.0.0:2380",
> > +                   initial_cluster_token="etcd-cluster-2",
> > +                   initial_cluster="calico=http://%s:2380" % self.ip,
> > +                   initial_cluster_state="new",
> > +                  )
> > +
> > +    def get_ip(self):
>
> actually the example i pasted was prolly a bit rubbish since it uses a
> name rather than an IP so not sure what happens if DNS isn't working
>
> I usually see this with a "fake" IP of "8.8.8.8" or similar - it doesn't
> actually need to be valid and so it's not actually talking to the outside
> world.
>
> @alexaltair <https://github.com/alexaltair> when you say it works for now
> - you mean it works for you because your environment uses eth0 or are you
> saying this should work for everyone?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Metaswitch/calico-docker/pull/154/files#r31346709>.
>